pkg/container-collection/container-collection.go

```diff
@@ -311,6 +325,31 @@ func (cc *ContainerCollection) Enrich(event *eventtypes.CommonData, mountnsid ui
 	}
 }
 
+func (cc *ContainerCollection) EnrichByNetNs(event *eventtypes.CommonData, netnsid uint64) {
+	event.Node = cc.nodeName
+
+	containers := cc.LookupContainersByNetns(netnsid)
+	if len(containers) == 0 || containers[0].HostNetwork {
+		return
+	}
+	if len(containers) == 1 {
+		event.Container = containers[0].Name
+		event.Pod = containers[0].Podname
+		event.Namespace = containers[0].Namespace
+		return
+	}
+	if containers[0].Podname != "" && containers[0].Namespace != "" {
+		// Kubernetes containers within the same pod.
+		// TBD: Should we verify that all containers belong to the same pod?
+		event.Pod = containers[0].Podname
+		event.Namespace = containers[0].Namespace
+	}
+	// else {
+	// 	TODO: Non-Kubernetes containers sharing the same network namespace.
+	// 	What should we do here?
+	// }
```

Comment:
Suggestion?

Reply:
Maybe leave TODO until we support all containers?

Reply:
We already support all containers (Kubernetes and non-Kubernetes). The use case here is to have more than one non-Kubernetes container sharing the same network namespace. In that case, we should not enrich the event with the container name of none because we are not sure which of them actually generated the event. And being non-Kubernetes containers, there will not be Kubernetes metadata (namespace / pod). Therefore, the event will have no reference at all.

I don't know what we can do here, we could show all the possible containers where the event could have been generated? But how to present it? Using the group feature of the columns library? ... I think we can manage this in another PR letter as it is a very specific use case.